src/librustc_resolve/resolve_imports.rs

```diff
@@ -52,7 +52,10 @@ pub enum ImportDirectiveSubclass<'a> {
         value_result: Cell<Result<&'a NameBinding<'a>, Determinacy>>,
         type_result: Cell<Result<&'a NameBinding<'a>, Determinacy>>,
     },
-    GlobImport { is_prelude: bool },
+    GlobImport {
+        is_prelude: bool,
+        max_vis: Cell<ty::Visibility>, // The visibility of the greatest reexport.
```

Comment:
No, this visibility doesn't matter during import resolution as you expect. It is only used in `finalize_import`s [here](https://github.com/rust-lang/rust/pull/35894/files#diff-9a6ea146fd5deda078e1d8402f5cf4eeR600) to enforce the requirement that a glob must import something with at least the glob's visibility.

It would also be sufficient to use `imports_name_with_sufficient_visibility: bool` instead of `max_vis: Cell<ty::Visibility>`, but I think knowing `max_vis` could be useful later for better error messages, for example;
```rust
mod foo {
    pub(crate) fn f() {}
}
pub use foo::*; //~ ERROR This glob must import something with `pub` visiblity
                //~^ NOTE consider using `pub(crate) use foo::*;`